prusti-interface/src/specs/typed.rs

```diff
@@ -203,6 +203,7 @@ pub struct ProcedureSpecification {
     pub pledges: SpecificationItem<Vec<Pledge>>,
     pub trusted: SpecificationItem<bool>,
     pub terminates: SpecificationItem<Option<LocalDefId>>,
+    pub purity: SpecificationItem<Option<DefId>>, // for type-conditional spec refinements
```

Comment:
I found this addition really strange. Why do we need another field just for purity? Isn't the refinability of `kind` already expressed in the `SpecificationItem` type?